docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+Calico etcd-based architecture
+==============================
+
+The current version of Calico is built around `etcd`_. etcd is a distributed,
+consistent key value store for shared configuration and service discovery with
+a focus on being simple, secure, fast, and reliable.
+
+In Calico, etcd is used as the data store and communication mechanism for all
+the Calico components. This data store contains all the information the various
+Calico components need to set up the Calico network.
+
+This document discusses the various pieces of the Calico etcd-based
+architecture, with a focus on what specific role each component plays in the
+Calico network. This does not discuss the Calico etcd data model, which also
+acts as the primary API into the Calico network: for more on that, see
+:doc:`etcd-data-model`.
+
+.. _etcd: https://github.com/coreos/etcd
+
+Components
+----------
+
+Calico is made up of the following interdependent components:
+
+- :ref:`calico-felix-component`, the primary Calico agent that runs on each
+  machine that hosts endpoints.
+- :ref:`calico-orchestrator-plugin`, orchestrator-specific code that tightly
+  integrates Calico into that orchestrator.
+- :ref:`calico-etcd-component`, running unmodified and used by the other
+  components as a data store.
```

Comment:
"running unmodified" feels like unwarranted detail here.  I'd prefer just "the data store".